Bottom: 0cf18140759f544d7939aa62d2647161675c47b4
Top:    08abd3b0f605ac12a943dd030b33e4bdb69e4afb
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-01 15:37:45 +0800

debug


---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 9a5bb8b..dd35dae 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -443,7 +443,7 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 
 			inc = check_master_self_include(master, ctxt);
 			if (inc) 
-				master->recurse = 1;;
+				master->recurse = 1;
 			master->depth++;
 			status = lookup_nss_read_master(master, age);
 			if (!status) {
@@ -451,6 +451,11 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 				     MODPREFIX
 				     "failed to read included master map %s",
 				     master->name);
+				/*
+				 * Plus map include failures don't cause the map
+				 * read to fail.
+				 */
+				master->read_fail = 0;
 			}
 			master->depth--;
 			master->recurse = 0;
